server/core/src/main/java/io/whitefox/core/types/predicates/BasePrimitiveTypeNames.java

```diff
@@ -0,0 +1,17 @@
+package io.whitefox.core.types.predicates;
+
+public enum BasePrimitiveTypeNames {
```

Comment:
can this be expressed as a subset of `io.whitefox.core.types.BasePrimitiveType`?